pkg/core/mempool/mem_pool.go

```diff
@@ -154,13 +159,61 @@ func (mp *Pool) Add(t *transaction.Transaction, fee Feer) error {
 		mp.lock.Unlock()
 		return ErrDup
 	}
+	if fee.P2PSigExtensionsEnabled() {
+		var (
+			txsFromMempool    []util.Uint256
+			txsFromAttributes []util.Uint256
+			ok                bool
+		)
+		// Step 1: check if `t` was in attributes of mempooled transactions
+		if txsFromMempool, ok = mp.conflicts[t.Hash()]; ok {
+			// check firstly if there are some mempooled transactions which have better network fee,
+			// if so, reject the new transaction
+			for _, hash := range txsFromMempool {
+				if mp.verifiedMap[hash].NetworkFee >= t.NetworkFee {
```

Comment:
We have to make an additional safety check here. Sender of `t` must be one of the signers of conflicting transaction. Consider scenario: Alice signs transaction A and sends it to the network, then Bob sees this transaction in the mempool and immediately creates transaction B conflicting with A and paying higher fee. Current rules would accept B in favor of A and if Bob has enough GAS he can effectively DoS Alice for as long as he wants to. So B has to be signed by Alice to prevent this, if it's not then it's invalid and A should stay in the pool irrespective of B's fee.